### ISSUE TITLE
TimeSeriesForecaster

### ISSUE THREAD

**Issue by haifeng-jin** [keras-team/autokeras#782]

**Comment by koernertim** [comment]
Has there been any development regarding this?

**Comment by haifeng-jin** [comment]
There are two possible approaches.
1. Put the preprocessing step inside the Keras model.
We need a preprocessing layer to change the shape.
The input shape to the preprocessing layer (num_instances, num_features).
The output shape of the preprocessing layer (num_instances - lookback + 1, lookback, num_features).
Is it possible to implement a preprocessing layer like this?

2. Do the preprocessing before input to the Keras model.
This can be directly done with input_adpaters.py.
The problem is when people export the Keras model, the model cannot do the preprocessing for them.
They have to do the preprocessing themselves to use the exported Keras model.
A possible solution would be to add one more argument for whether to do the preprocessing for the user in the fit.
They can choose to do preprocessing themselves when using AutoKeras.

**Comment by haifeng-jin** [comment]
@abgese 
I have got a reply confirming we cannot implement a preprocessing layer doing the job. Let's take the second approach to add a TimeSeriesInput() in nodes.py and a corresponding adapter in input_adapters.py.

Let me know if you have any questions.

We may use this feature later when it is in TensorFlow's stable release.
https://github.com/keras-team/governance/blob/master/rfcs/20190729-keras-preprocessing-redesign.md#timeseriesgenerator

**Comment by haifeng-jin** [comment]
@abgese I have merged the PR. I just found that there are still 2 minor changes we need to make.

1. We should change the default value of `predict_until`. Since we can know how long the user want to predict based on how long is the training and testing `x`. We just predict until the end of the `x`.

2. I am a little bit confused about why we don't need `y` in the `predict` function. I think the previous `y`s are also part of the input to predict the next step.

Thanks.
